### ISSUE TITLE
dockview setTitle method

### ISSUE THREAD

**Issue by mathuo** [mathuo/dockview#189]
Doesn't seem to be working anymore. Example:

```typescript
const Component = (props?: IGridviewPanelProps) => {
   
    useEffect(() => {
        const interval = setInterval(() => {

            console.log('component_id', global.layout.getPanel(props.api.id).api.title); //works
            global.layout.getPanel(props.api.id).api.setTitle("ASDASDASD"); // not working, no error given

        }, 1000);

        return () => {
            clearInterval(interval);
        };
    }, []);

    return <ComponentChild/>;
};
```

On the other hand, this piece of code changes the title but I got a TS error:

`global.layout.getPanel(props.api.id).setTitle("ASDASDASD");`

`TS2339: Property 'setTitle' does not exist on type 'IDockviewPanel'.`

_Originally posted by @amirmamaghani in https://github.com/mathuo/dockview/issues/5#issuecomment-1438800750_

**Comment by mathuo** [comment]
`setTitle` is a method that exists on the DockviewPanel object, however it is intended to be a private method now. The reason being is that all interactions should be via the `panel.api` to simplify the logic going forwards.

The intended behaviour is that the user calls `panel.api.setTitle(...)` however there seems to be a bug with this approach which I will fix.

For now though you can still set the title via an alternative approach which is a rather generic method exposed for advanced cases, but it does provide a way to update the title. Once fixed though it would be better to set this via `panel.api.setTitle(...)`

```
panel.update({ params: { title: 'my_title' } });
```

**Comment by mathuo** [comment]
Fixed in release 1.6.0

**Comment by ronen25** [comment]
Hi, first of all - thank you so much for your work!

It appears that in version 1.14.1 neither one of the suggested solutions work.
**However**, using a combination of both `update` and `setTitle` work, for example:

```
      const panel = containerApi.getPanel(panelIdToGet);

      panel?.update({ params: { title: titleToSet } });
      panel?.api.setTitle(titleToSet);
```

We're using React 18.2 with Vite 4.5.2 if that helps.

Thanks again! This library is a blast to use!

**Comment by mathuo** [comment]
Fix released in `1.14.2`. 

See https://dockview.dev/docs/core/panels/add for an example of title updating.

Title should be updated via `api.setTitle(...)`
